python/lsst/daf/butler/script/queryCollections.py

```diff
@@ -21,20 +21,160 @@
 
 from __future__ import annotations
 
-import itertools
-from typing import Iterable, Union
+from typing import Iterable
 
 from astropy.table import Table
-from numpy import array
 
 from .._butler import Butler
 from ..registry import CollectionType
 
 
+def _getTable(
+    repo: str,
+    glob: Iterable[str],
+    collection_type: Iterable[CollectionType],
+    inverse: bool,
+) -> Table:
+    """Run queryCollections and return the results in Table form.
+
+    Only lists the first child (or parent if `inverse` is `True`) in the
+    description column.
+
+    Parameters
+    ----------
+    repo
+    glob
+    collection_type
+        Same as `queryCollections`
+    inverse : `bool`
+        True if parent CHAINED datasets of each dataset should be listed in the
+        description column, False if children of CHAINED datasets should be
+        listed.
+
+    Returns
+    -------
+    collections : `astropy.table.Table`
+        Same as `queryCollections`
+    """
+    typeCol = "Type"
+    descriptionCol = "Parents" if inverse else "Children"
+    table = Table(
+        names=("Name", typeCol, descriptionCol),
+        dtype=(str, str, str),
+    )
+    butler = Butler(repo)
+    names = list(
+        butler.registry.queryCollections(collectionTypes=frozenset(collection_type), expression=glob or ...)
+    )
+    if inverse:
+        for name in sorted(names):
+            type = butler.registry.getCollectionType(name).name
+            parentNames = butler.registry.getCollectionParentChains(name)
+            if parentNames:
+                first = True
+                for parentName in sorted(parentNames):
+                    table.add_row((name if first else "", type if first else "", parentName))
+                    first = False
+            else:
+                table.add_row((name, type, ""))
+        # If none of the datasets has a parent dataset then remove the
+        # description column.
+        if not any(c for c in table[descriptionCol]):
+            del table[descriptionCol]
+    else:
+        for name in sorted(names):
+            type = butler.registry.getCollectionType(name).name
```

Comment:
Instead of converting the enum object to a string here, I'd recommend leaving it, using `if type is CollectionType.CHAINED` in the `if` block on the next line, and then using `.name` only when you need a string (e.g. in `table.add_row`).